### PR TITLE
Review and improve test coverage

### PR DESCRIPTION
**Pull request by enyst**
- Add tests for OpenHandsViewProvider (31.25% → 100%)
- Enhance SettingsManager tests for branch coverage (35.48% → 100%)
- Add 15 new tests covering AWS credentials, LLM fields, edge cases
- Overall coverage improved: 85.46% → 87.21% statements, 78.21% → 82.92% branches

Tests added:
- OpenHandsViewProvider: 8 tests for TreeDataProvider, commands, icons
- SettingsManager: 7 tests for AWS secrets, optional LLM fields, sanitization, null handling

All 164 tests passing. Issue #62 related.

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for settings management and validation.
  * Added test coverage for sidebar navigation and tree view functionality.

### CODE REVIEW

File: src/settings/__tests__/SettingsManager.test.ts [hunk 1]
```diff
@@ -59,4 +59,145 @@ describe('SettingsManager', () => {
...
+    const originalGet = a.get.bind(a);
+    a.get = <T>(key: string, def?: T): T | undefined => {
+      if (key === 'openhands.serverUrl') return null as any;
+      if (key === 'openhands.agent.enableSecurityAnalyzer') return null as any;
+      if (key === 'openhands.conversation.maxIterations') return null as any;
+      if (key === 'openhands.confirmation.policy') return null as any;
+      if (key === 'openhands.confirmation.risky.threshold') return null as any;
+      if (key === 'openhands.confirmation.risky.confirmUnknown') return null as any;
+      return originalGet(key, def);
+    };
```
[the remark this answers is not in the document]
Comment (enyst) [reply src/settings/__tests__/SettingsManager.test.ts:170]: /gemini wdyt now?

File: src/sidebar/__tests__/OpenHandsViewProvider.test.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    });
+
+    it('getTreeItem returns the element unchanged', () => {
+      const mockItem = new vscode.TreeItem('Test Item');
+      const result = provider.getTreeItem(mockItem as any);
+      expect(result).toBe(mockItem);
+    });
+  });
+
+  describe('getChildren', () => {
```
[the remark this answers is not in the document]
Comment (enyst) [reply src/sidebar/__tests__/OpenHandsViewProvider.test.ts:24]: /gemini did we fix this?
